[FLINK-5751] [docs] Add link check script

uce opened the pull request:
This is a slightly adjusted script from @patricklucas as posted in FLINK-5751.

I was wondering whether it makes sense to have it inside the repo, the local work flow would be:
```sh
./build_docs.sh -p
./check_links.sh
```

You can overwrite the URL to check like `./check_links.sh https://ci.apache.org/projects/flink/flink-docs-master/`.

@rmetzger @patricklucas Should we include this?

patricklucas commented on docs/check_links.sh:
```diff
@@ -0,0 +1,39 @@
+################################################################################
+
+# Don't abort on any non-zero exit code
+#set +e
```
This is a vestige from me writing this as a Jenkins jobs; you can remove this line and the above comment.

patricklucas commented on docs/check_links.sh:
```diff
@@ -0,0 +1,39 @@
+target=${1:-"http://localhost:4000"}
+
+# Crawl the docs, ignoring robots.txt, storing nothing locally
+wget --spider -r -nd -nv -e robots=off -p -o spider.log $target
```
$target in quotes

patricklucas commented on docs/check_links.sh:
```diff
@@ -0,0 +1,39 @@
+
+# Fail the build if any broken links are found
+broken_links_str=$(grep -e 'Found [[:digit:]]\+ broken link' spider.log)
+echo -e "$broken_links_str"
```
If you're going to get rid of my pretty bold-red formatting you can drop the `-e` too. :)

Actually, might as well put the `echo` line inside the `if` too.

uce commented:
Thanks @patricklucas, I've addressed your comments and re-added the red formatting, which is nice indeed. ;-) I'm going to merge this with you as the author. Hope that's fine.

patricklucas commented:
:+1:

patricklucas commented on docs/check_links.sh:
```diff
@@ -0,0 +1,36 @@
+# Fail the build if any broken links are found
+broken_links_str=$(grep -e 'Found [[:digit:]]\+ broken link(s)' spider.log)
+if [ -n "$broken_links_str" ]; then
+	echo -e "\e[1;31m$broken_links_str\e[0m"
```
Mixed tabs/spaces in this file if you want to do a no-review follow-up commit.

uce replied:
Thanks for catching this! Fixed in a separate commit.
